module/VuFind/tests/integration-tests/src/VuFindTest/Mink/LoggingTest.php

```diff
@@ -0,0 +1,267 @@
+                    '/VuFind Log Alert/',
+                    '/CRITICAL:/',
+                    '/404 Not Found/',
+                    '/RequestErrorException/',
+                    '/VuFindSearch\\\\Backend\\\\Exception/',
+                    '/Search\/Results.*lookfor.*test/',
+                ],
+                'unexpected_patterns' => [
+                    '/DEBUG:/',
+                    '/INFO:/',
+                ],
+                'min_emails'          => 1,
+                'description'         => 'Should log critical errors when Solr connection fails',
+            ],
+            'debug_logging_only'      => [
+                'email_config'        => 'debug@myuniversity.edu:debug-5',
+                'expected_patterns'   => [
+                    '/VuFind Log Alert/',
+                    '/DEBUG:/',
+                ],
+                'unexpected_patterns' => [
+                ],
+                'min_emails'          => 1,
+                'description'         => 'Should capture debug messages when debug logging is enabled',
+            ],
+            'minimal_detail_level'    => [
+                'email_config'        => 'alerts@myuniversity.edu:error-1',
+                'expected_patterns'   => [
+                    '/VuFind Log Alert/',
+                    '/CRITICAL:/',
+                    '/404 Not Found/',
+                ],
+                'unexpected_patterns' => [
+                    '/Backtrace:/',
+                    '/Server Context:/',
+                    '/Array/',
+                ],
+                'min_emails'          => 1,
+                'description'         => 'Should provide minimal detail at level 1',
+            ],
+            'maximum_detail_level'    => [
+                'email_config'        => 'alerts@myuniversity.edu:error-5',
+                'expected_patterns'   => [
+                    '/VuFind Log Alert/',
+                    '/CRITICAL:/',
+                    '/404 Not Found/',
+                    '/Backtrace:/',
+                    '/Server Context:/',
+                    '/HTTP_USER_AGENT/',
+                    '/REQUEST_URI/',
+                ],
+                'unexpected_patterns' => [],
+                'min_emails'          => 1,
+                'description'         => 'Should provide maximum detail at level 5',
+            ],
+        ];
+    }
+
+    /**
+     * Test email logging functionality with various configurations
+     *
+     * @param string $emailConfig        Email configuration string
+     * @param array  $expectedPatterns   Patterns that should be found in log
+     * @param array  $unexpectedPatterns Patterns that should NOT be found in log
+     * @param int    $minEmails          Minimum number of emails expected
+     * @param string $description        Test scenario description
+     *
+     * @return void
+     *
+     * @dataProvider loggingScenarioProvider
+     */
+    public function testLogging(
```

Comment:
```suggestion
    public function testEmailLogging(
```

I'd also suggest changing `loggingScenarioProvider` to `emailLoggingScenarioProvider`.